File: poo/python/Interfaces/Conta/CDB.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 12/04/14
"""
from __future__ import unicode_literals
from __future__ import print_function

from Conta import Conta

class CDB(Conta):

    # Construtor #
    def __init__(self, numero):
        super(CDB, self).__init__(numero)


    def atualiza(self, taxa):
        pai = super(CDB, self)
        pai.setSaldo(pai.getSaldo() + 2.5 * pai.getSaldo() * taxa)


    def calculaTributos(self):
        return 42

File: poo/java/Static/Exercicio4/CDB.java
/*
 *  Programa escrito com propósito didático apenas.
 *  @author Nathan Cruz <nathan.cruz@unifesp.br>
 *  @created 04/04/14
 *  R.A.: 86492
 */

public class CDB extends Conta implements Tributavel{

    /* Construtor */
    public CDB(int numero, String nome){
        this.numero = numero;
        this.nome = nome;
    }

    @Override
    public void atualiza(double taxa){
        this.saldo += 2.5*(this.saldo * taxa);
    }

    /* Implementacao da interface */
    public double calculaTributos(){
        return 42;
    }

    @Override
    public String toString(){
        return super.toString() +
               "\n - Tipo de Conta: CDB";
    }
}
